### PR TITLE
Add app change URL feature

### PR DESCRIPTION
**Pull request by JimboJoe**
## Problems
App URL change feature is not available yet from the admin interface.

## Solution
Add a new "Change URL" button in the app info page.

## How to test
Can be tested with any app that has implemented the `change_url` script (see [here](https://pr.yunohost.org/appci.html))

## PR Status
Opinion needed

## Validation

- [x] Principle agreement 0/2 : 
- [x] Quick review 0/1 : 
- [x] Simple test 0/1 : 
- [ ] Deep review 0/1 :

### CODE REVIEW

File: src/js/yunohost/controllers/apps.js [hunk 1]
```diff
@@ -572,4 +572,50 @@
...
+                  // Pre-fill with current path
+                  path: app_data.settings.path
+                };
+                c.view('app/app_changeurl', data);
+            });
+        });
+    });
+    
+    // Change app URL
+    app.post('#/apps/:app/changeurl/submit', function (c) {
```
Comment (opi) [inline src/js/yunohost/controllers/apps.js]: you can probably use the same route (`#/apps/:app/changeurl`).

Comment (JimboJoe) [reply src/js/yunohost/controllers/apps.js]: Indeed, that works!

File: src/views/app/app_info.ms [hunk 2]
```diff
@@ -49,6 +49,13 @@
         </div>
         <hr>
         <div class="container">
+            <p>{{t 'app_info_changeurl_desc' settings.domain}}</p>
```
Comment (opi) [inline src/views/app/app_info.ms]: this part will be displayed even if the app does not provide a change_url script ?

Comment (JimboJoe) [reply src/views/app/app_info.ms]: Yes. Unfortunately, the API doesn't provide a way to know if it's available... But I agree we should ideally disable this.

Comment (opi) [reply src/views/app/app_info.ms]: so what happen if I try to use this feature with an app that does not have a `change_url` script ?

Comment (JimboJoe) [reply src/views/app/app_info.ms]: You immediately get an explicit error message (from moulinette) in the log section:

Comment (Psycojoker) [reply src/views/app/app_info.ms]: I'm going to look if I can add this in the API.

Comment (Psycojoker) [reply src/views/app/app_info.ms]: Done https://github.com/YunoHost/yunohost/pull/402

**Comment by JimboJoe** [comment]
Now using https://github.com/YunoHost/yunohost/pull/402 to display the "Change URL" button only if the feature is available (thanks @Psycojoker !).
Tested and ready for review.

**Comment by JimboJoe** [comment]
I would really like that PR to be over, but I had a though: I implemented it so that the "Change URL" button only appears if the feature has been implemented. Maybe that could be unclear to users whether it's a bug, and why it is application-dependent.
Wouldn't having only the button disabled with an explicit tooltip (e.g. "this feature hasn't been implemented in this app yet') be better...?

**Comment by Psycojoker** [comment]
> Wouldn't having only the button disabled with an explicit tooltip (e.g. "this feature hasn't been implemented in this app yet') be better...?

I think yes and that would make the user knowledgeable of this potential feature and motivate applications developpers to implement it.

**Comment by JimboJoe** [comment]
Needed a bit of work to add Handlebars helpers to use Bootstrap's tooltips.
Final result (when hovering the disabled button):


Ready for review.

**Pull request merged**